demos/common/cpp/utils/src/image_utils.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+// Copyright (C) 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "utils/image_utils.h"
+
+cv::Mat resizeImageExt(const cv::Mat& mat, int width, int height, RESIZE_MODE resizeMode, bool hqResize, cv::Rect* roi) {
+    if (width == mat.cols && height == mat.rows) {
+        return mat;
+    }
+
+    cv::Mat dst;
+    int interpMode = hqResize ? cv::INTER_LINEAR : cv::INTER_CUBIC;
+
+    switch (resizeMode) {
+    case RESIZE_FILL:
+    {
+        cv::resize(mat, dst, cv::Size(width, height), interpMode);
+        if (roi) {
+            *roi = cv::Rect(0, 0, width, height);
+        }
+        break;
+    }
+    case RESIZE_KEEP_ASPECT:
```

Comment:
The `RESIZE_KEEP_ASPECT` case implementation is the same as `RESIZE_KEEP_ASPECT_LETTERBOX`, the only difference is that `dx = 0, dy = 0`, so the whole resizing should not be repeated. It can be combined with `if ... else` condition

Reply:
good catch ! I've changed the code